recipes/dns.rb

```diff
@@ -9,6 +9,8 @@
 template '/etc/dnsmasq.d/dns.conf' do
   source 'dynamic_config.erb'
   mode 0644
+  retries 3
+  retry_delay 3
```

Comment:
Instead of using retries, it's probably better to have the resource notification be delayed instead of immediately.